src/uct/api/uct.h

```diff
@@ -680,6 +697,76 @@ int uct_iface_is_reachable(uct_iface_h iface, const uct_iface_addr_t *addr);
 
 /**
  * @ingroup UCT_RESOURCE
+ * @brief Create an event handle for interrupt notification.
+ *
+ * @param [in]  iface       Handle to an open communication interface.
+ * @param [in]  events      Requested event types for notification.
+ * @param [out] wakeup_p    Location to write the notification event handle.
+ *
+ * @return Error code.
+ */
+ucs_status_t uct_wakeup_open(uct_iface_h iface, unsigned events,
+                             uct_wakeup_h *wakeup_p);
+
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Obtain a notification file descriptor for polling.
+ *
+ * @param [in]  wakeup     Handle to the notification event.
+ * @param [out] fd         Location to write the notification file descriptor.
+ *
+ * @return Error code.
+ */
+ucs_status_t uct_wakeup_efd_get(uct_wakeup_h wakeup, int *fd_p);
+
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Request the next notification on the event.
+ *
+ * @param [in] wakeup      Handle to the notification event.
+ *
+ * @return Error code.
+ */
+ucs_status_t uct_wakeup_efd_arm(uct_wakeup_h wakeup);
+
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Wait for the next notification.
+ *
+ * @param [in] wakeup      Handle to the notification event.
+ *
+ * @return Error code.
+ */
+ucs_status_t uct_wakeup_wait(uct_wakeup_h wakeup);
+
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Cause the next notification.
+ *
+ * @param [in] wakeup      Handle to the notification event.
+ *
+ * @return Error code.
+ */
+ucs_status_t uct_wakeup_signal(uct_wakeup_h wakeup);
+
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Close the notification event handle.
+ *
+ * @param [in] wakeup      Handle to the notification event.
+ *
+ * @return Error code.
+ */
+ucs_status_t uct_wakeup_close(uct_wakeup_h wakeup);
```

Comment:
the return value should be void
also pls move it to be right after uct_wakeup_open